yjit/src/codegen.rs

```diff
@@ -834,20 +836,20 @@ pub fn gen_single_block(blockid: BlockId, start_ctx: &Context, ec: EcPtr, cb: &m
     Ok(blockref)
 }
 
-fn gen_nop(jit: &mut JITState, ctx: &mut Context, cb: &mut CodeBlock, ocb: &mut OutlinedCb) -> CodegenStatus
+fn gen_nop(_jit: &mut JITState, _ctx: &mut Context, _cb: &mut CodeBlock, _ocb: &mut OutlinedCb) -> CodegenStatus
 {
     // Do nothing
     KeepCompiling
 }
```

Comment:
I'm generally onboard, but the changes in `codegen.rs` look pretty tedious. We have all these functions with the same signature, and then you kind of have to individually annotate the signatures to say which parameters are unused. That seems kinda icky, particularly considering that it's 100% fine for these functions not to use any of their parameters.